File: rental_car_angular/src/app/modules/admin/components/admin.service.ts
import { Injectable } from '@angular/core';
<<<<<<< HEAD
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';
const BASIC_URL = ["http://localhost:8080"]
=======
>>>>>>> c8b8b084b4274a4ac949778bdc28d467cc13cd75

@Injectable({
  providedIn: 'root'
})
export class AdminService {

<<<<<<< HEAD
  constructor(private http:HttpClient) { }

  postCar(carDto : any): Observable<any>{
    return this.http.post(BASIC_URL + '/api/admin/car', carDto, {
      headers: this.createAuthorizationHeader()
    });
  }

  deleteCar(id : number): Observable<any>{
    return this.http.delete(BASIC_URL + "/api/admin/car/" + id ,{
    headers : this.createAuthorizationHeader()
  });
  }

  getCarById(id : number): Observable<any>{
    return this.http.get(BASIC_URL + "/api/admin/car/" + id ,{
      headers : this.createAuthorizationHeader()
    });
  }

  createAuthorizationHeader() : HttpHeaders{
    let authHeaders : HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization', 
      'Bearer ' + StorageService.getToken()
    );
  }


=======
  constructor() { }
>>>>>>> c8b8b084b4274a4ac949778bdc28d467cc13cd75
}

File: src/main/java/com/CarRental/Car/services/admin/AdminService.java
package com.CarRental.Car.services.admin;

import java.io.IOException;
import java.util.List;

import com.CarRental.Car.CarDto;
import com.CarRental.Car.CarDtoListDto;
import com.CarRental.Car.SearchCarDto;

public interface AdminService {

    boolean postCar(CarDto carDto) throws IOException;

    List<CarDto> getAllCars();

    void deleteCar(Long id);

    CarDto getCarById(Long id);

    boolean updateCar(Long carId , CarDto carDto) throws IOException;
    
    CarDtoListDto searchCar(SearchCarDto searchCarDto);

}
